### PR TITLE
Qt: Fix data race and undefined behaviour in games_config, fix premature futurewatcher cancelation

### LINKED ISSUE #13767: [Regression] Rpcs3 displays only PS3 Interface XMV in game lista frer i boot software on  0.0.27-14930
### Quick summary

 Rpcs3 displays only PS3 Interface XMB in game list when i boot software on 0.0.27-14930
It doesn't happen on 14845

### Details

image
image

### Build with regression

0.0.27-14930

### Attach two log files

broken
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11336547/RPCS3.log)
working
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11336554/RPCS3.log)

### Attach capture files for visual issues

_No response_

### System configuration

_No response_

### Other details

_No response_

### PR DESCRIPTION
**Pull request by Megamouse**
- Fixes a potential data race and undefined behaviour when iterating over games.yml content
- Fixes premature futurewatcher cancelation which led to the abortion of the directory parser (isStarted is basically always true)

fixes #13767

### CODE REVIEW

File: rpcs3/Emu/games_config.cpp [hunk 1]
```diff
@@ -64,6 +74,8 @@ bool games_config::add_game(const std::string& key, const std::string& path)
 
 bool games_config::save()
 {
+	std::lock_guard lock(m_mutex);
```
Comment (elad335) [inline rpcs3/Emu/games_config.cpp:77]: save shouldn't occur cuncurrently, I siggest using trylock() here and if it fails return prematurely and log an error.

**Review by elad335 — CHANGES_REQUESTED** [review]
I suggest to use our atomic_ptr to copy values and store a new one (changed value) by making a new shared_ptr, header is at util/shared_ptr.hpp.  This way a mutex won't be required.

File: rpcs3/Emu/games_config.cpp (hunk 1 above)
Comment (Megamouse) [reply rpcs3/Emu/games_config.cpp:77]: There is a Szenario where the UI is refreshed while a game triggers a boot.
(Whether or not this happens on the main thread anyway is not relevant due to future proofing)

**Comment by Megamouse** [comment]
I'll merge this as hot fix for the game list

File: rpcs3/Emu/games_config.cpp (hunk 1 above)
Comment (elad335) [reply rpcs3/Emu/games_config.cpp:77]: Which is clearly invalid they cannot coexist.

Comment (elad335) [reply rpcs3/Emu/games_config.cpp:77]: The decision to use AddGamesFromDir for /games was a mistake.

Comment (Megamouse) [reply rpcs3/Emu/games_config.cpp:77]: It doesn't matter with the other PR

Comment (Megamouse) [reply rpcs3/Emu/games_config.cpp:77]: It's a rare edge case that should never happen

**Pull request merged**